src/rpc/misc.cpp

```diff
@@ -501,50 +499,46 @@ bool getAddressFromIndex(const AddressType type, const uint160 &hash, std::strin
     }
 }
 
-bool getAddressesFromParams(const UniValue& params, std::vector<std::pair<uint160, AddressType> > &addresses)
-{
-    if (params[0].isStr()) {
-        CBitcoinAddress address(params[0].get_str());
-        uint160 hashBytes;
-        AddressType type = AddressType::UNKNOWN;
-        if (!address.GetIndexKey(hashBytes, type)) {
-            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid address");
-        }
-        addresses.push_back(std::make_pair(hashBytes, type));
-    } else if (params[0].isObject()) {
+std::pair<uint160,AddressType> addressHashAndTypeFromValue(const UniValue& value) {
+    const std::string value_str{value.get_str()}; // Throws if not string type
 
-        UniValue addressValues = find_value(params[0].get_obj(), "addresses");
-        if (!addressValues.isArray()) {
-            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Addresses is expected to be an array");
-        }
+    uint160 hashBytes;
+    AddressType type = AddressType::UNKNOWN;
+    CBitcoinAddress address(value_str);
+    if (!address.GetIndexKey(hashBytes, type)) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, std::string("Invalid address : ") + value_str);
+    }
+    return {hashBytes, type};
+}
 
-        std::vector<UniValue> values = addressValues.getValues();
+std::vector<std::pair<uint160, AddressType>> addressesHashAndTypeFromValue(const UniValue& value, bool allow_empty = true) {
 
-        for (std::vector<UniValue>::iterator it = values.begin(); it != values.end(); ++it) {
+    // TODO See Jira's discussion ZEND-242
+    if (value.isStr()) {
+        return {addressHashAndTypeFromValue(value)};
+    }
 
-            CBitcoinAddress address(it->get_str());
-            uint160 hashBytes;
-            AddressType type = AddressType::UNKNOWN;
-            if (!address.GetIndexKey(hashBytes, type)) {
-                throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid address");
-            }
-            addresses.push_back(std::make_pair(hashBytes, type));
-        }
-    } else {
-        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid address");
+    if (!value.isArray() || (value.empty() && !allow_empty)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, std::string("An array of addresses is expected") + (allow_empty ? "" : " [not empty]"));
     }
 
-    return true;
+    std::vector<std::pair<uint160, AddressType>> ret;
+    ret.reserve(value.size());
+    for (const auto& item : value.getValues()) {
+        ret.push_back(addressHashAndTypeFromValue(item));
+    }
+    return ret;
 }
 
-#ifdef ENABLE_ADDRESS_INDEXING
```

Comment:
I think the specification _"(requires addressindex to be enabled)"_ should be dropped from the help-documentation of the following methods:

- `getaddressmempool`
- `getaddressutxos`
- `getaddressdeltas`
- `getaddressbalance`
- `getaddresstxids`